trappy/cpu_power.py

```diff
@@ -120,6 +143,89 @@ def get_all_freqs(self, mapping_label):
 
         return pivot_with_labels(dfr, "freq", "cpus", mapping_label) / 1000
 
+    def plot_cdev_states(self, title="", width=None, height=None, xlim="default",
+                         ylim="range", legend_label="", cpus=None):
+        """Plot the cooling device state evolution
+
+        :param ax: Axis instance
+        :type ax: :mod:`matplotlib.Axis`
+
+        :param legend_label: Label for the legend
+        :type legend_label: str
+
+        :param title: The title of the plot
+        :type title: str
+
+        :param width: The width of the plot
+        :type width: int
+
+        :param height: The height of the plot
+        :type height: int
+
+        :param cpus: List of cpus to plot
+            All are plotted by default
+        :type tz_id: list of int
+        """
+        from matplotlib import pyplot as plt
+        from trappy.plot_utils import normalize_title, pre_plot_setup, post_plot_setup
+
+        if len(self.data_frame) == 0:
+            raise ValueError("Empty DataFrame")
+
+        thermal_dfr = self.data_frame.copy()
+
+        # Sanitize cpumasks
+        thermal_dfr["cpus"] = thermal_dfr["cpus"].apply(sanitize_mask)
+
+        # Find available cpumasks
+        available_masks = []
+        for mask in thermal_dfr["cpus"].unique().tolist():
+            available_masks.append(mask)
+
+        # Sanitize cpus
+        if cpus is not None:
+            global_mask = cpus_to_mask(cpus)
+
+            # Find masks that match the requested CPUs
+            # This can include other CPUs
+            selected_masks = []
+            for mask in available_masks:
+                if global_mask & mask:
+                    selected_masks.append(mask)
```

Comment:
You can shorten this with a list comp too: `select_masks = [m for m in available masks if m & global_mask]`